### PR TITLE
Implement partitioning map/reduce step

### PR DESCRIPTION
**Pull request by delucchi-cmu**
- methods to perform map reduce (map step co-occurs with histogram generation)
- do a better job with temp directories
- adds a bunch more test input data
- fix indexing in some test input data
- add progress reporting in single and dask runners

This *looks* like a bigger change than it is: the test input files are boring, and there are a lot of indentation changes to make unit test temp directories cleaned up after the test is finished.

### CODE REVIEW

File: src/partitioner/dask_runner.py [hunk 1]
```diff
@@ -1,24 +1,43 @@
-"""Partitioner runner that uses dask for parallelization"""
+"""Partitioner runner that doesn't use any parallelization mechanism"""
```
Comment (maxwest-uw) [inline src/partitioner/dask_runner.py]: kind of confused about what this means, seems like we still are using dask and maybe I'm just ignorant but wouldn't map reduce be considered a type of parallelization?

**Review by maxwest-uw — COMMENTED** [review]
one v small comment, but everything looks good!
also, on monday could you send me what data you're using to test and the process you're doing for manual testing? there've been two pretty large PRs in the past week and I'd like to play around with the code a bit myself in it's current state : )

File: src/partitioner/dask_runner.py (hunk 1 above)
Comment (delucchi-cmu) [reply src/partitioner/dask_runner.py]: That looks like an unfortunate copy/paste problem. This runner definitely uses parallelism.

**Review by maxwest-uw — APPROVED** [review]
lgtm, and thanks for sending me the run info 🤙

**Pull request merged**